src/core/error/common_error.rs

```diff
@@ -497,6 +497,53 @@ pub enum CommonError {
         "Transaction Guarantee's 'instruction_index' is out of bounds, the provided manifest contains #{count}, but an 'instruction_index' of {index} was specified."
     )]
     TXGuaranteeIndexOutOfBounds { index: u64, count: u64 } = 10138,
+
+    #[error("Failed to create KeyAgreementPublicKey from hex: {bad_value}")]
+    InvalidKeyAgreementPublicKeyFromHex { bad_value: String } = 10139,
+
+    #[error(
+        "Failed to create KeyAgreementPublicKey from bytes: {bad_value:?}"
+    )]
+    InvalidKeyAgreementPublicKeyFromBytes { bad_value: BagOfBytes } = 10140,
+
+    #[error(
+        "Failed to create KeyAgreementPrivateKey from bytes: {bad_value:?}"
+    )]
+    InvalidKeyAgreementPrivateKeyFromBytes { bad_value: BagOfBytes } = 10141,
+
+    #[error("Failed to expand HKDF, reason: '{underlying}'")]
+    HkdfExpandFailed { underlying: String } = 10142,
```

Comment:
This is not needed, cannot fail. Hmm, see:

```rust
impl VersionedPasswordBasedKeyDerivation for PbHkdfSha256 {
```

Reply:
no need to offset all errors if you dont wanna, you can rename this `FREE2 = 10142,`